service/cachemdw/middleware.go

```diff
@@ -0,0 +1,109 @@
+package cachemdw
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/kava-labs/kava-proxy-service/decode"
+)
+
+type cacheContextKey string
+
+const (
+	CachedContextKey cacheContextKey = "X-KAVA-PROXY-CACHED"
+
+	CacheHeaderKey       = "X-Cache"
```

Comment:
do we want this to be prefixed like the others like `"X-KAVA-PROXY-CACHE"`?

Reply:
yup

Reply:
done